compiler/testSuite/semtype/table-t.bal

```diff
@@ -0,0 +1,21 @@
+type R record {|
+    int id;
+    string f;
+|};
+
+type R1 record {|
+    int id;
+    string f;
+    float d;
+|};
+
+type READ readonly;
+
+// @type W < T
+// @type W < READ
+// @type T < READ
+// @type W < Y
+// @type Y <> T
+type T table<R> & readonly | table<R1> & readonly;
```

Comment:
We need a test case where there is a subtype relationship between the records (e.g. same fields but one member type is a subtype of the other)